### PR TITLE
Update Database Accesses

### LINKED ISSUE #57: Provide proper solution for database schemas
Code like `await leaderboardCursor.toArray() as LeaderboardSchema[];` should be able to be inferred using schemas. Schemas should be predefined with json schemas. Types can be then generated and everything in theory is type safe and mongodb safe.

### PR DESCRIPTION
**Pull request by AydanPirani**
Fix database accesses to work with Mongoose, and change formatting!

### CODE REVIEW

File: src/database/attendee-db.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public _id: string;
+
+    @prop({ required: true })
+    public userId: string;
+
+    @prop({ required: true })
+    public foodWave: number;
+
+    @prop({ required: true })
+    public score: number;
```
Comment (lasyaneti) [inline src/database/attendee-db.ts]: points?

Comment (lasyaneti) [reply src/database/attendee-db.ts]: if this is points - it should be part of profile, not metadata

Comment (AydanPirani) [reply src/database/attendee-db.ts]: p-oh-tato poh-ta-to. any preferences?

File: src/database/event-db.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    @prop({ required: true })
+    public description: string;
+
+    @prop({
+        required: true,
+        type: () => {
+            return String;
+        },
+    })
+    public tags: string[];
```
Comment (lasyaneti) [inline src/database/event-db.ts:19]: ngl idk what tags r for but we should prob keep it

File: src/constants.ts [hunk 3]
```diff
@@ -45,30 +39,19 @@ abstract class Constants {
...
 
     static readonly DEFAULT_JWT_OFFSET: string = "48h";
 
     // Constants for general usage
     static readonly ZERO: number = 0;
-    static readonly EVENT_ID_LENGTH: number = 24;
+    static readonly EVENT_ID_LENGTH: number = 32;
+    static readonly EVENT_BYTES_GEN: number = 16;
     static readonly MILLISECONDS_PER_SECOND: number = 1000;
     static readonly DEFAULT_POINT_VALUE: number = 10;
```
Comment (lasyaneti) [inline src/constants.ts]: should be 0

Comment (lasyaneti) [reply src/constants.ts]: unless it was 10 in the old api

Comment (AydanPirani) [reply src/constants.ts]: it wasn't, i can reset to 10

File: src/database/attendee-db.ts (hunk 1 above)
Comment (AydanPirani) [reply src/database/attendee-db.ts]: i thought metadata was data that we change, and profile data is data that the user changes

Comment (lasyaneti) [reply src/database/attendee-db.ts]: yes, points.

**Comment by AydanPirani** [comment]
Ready to merge whenever yall are!

File: src/constants.ts [hunk 4]
```diff
@@ -39,9 +39,10 @@ abstract class Constants {
         [this.DEV_DEVICE, this.DEV_REDIRECT],
     ]);
 
-    static readonly GITHUB_OAUTH_CALLBACK: string = "https://adonix.hackillinois.org/auth/github/callback/";
+    static readonly GITHUB_OAUTH_CALLBACK: string = "http://localhost:3000/auth/github/callback/";
+    // static readonly GITHUB_OAUTH_CALLBACK: string = "https://adonix.hackillinois.org/auth/github/callback/";
     // static readonly GOOGLE_OAUTH_CALLBACK: string = "https://adonix.hackillinois.org/auth/google/callback/";
-    static readonly GOOGLE_OAUTH_CALLBACK: string = "http://localhost:3000/auth/google/callback/";
+    static readonly GOOGLE_OAUTH_CALLBACK: string = "http://127.0.0.1:3000/auth/google/callback/";
```
Comment (lasyaneti) [inline src/constants.ts]: bruh

**Review by lasyaneti — APPROVED** [review]

**Pull request merged**